datafusion/spark/src/function/string/length.rs

```diff
@@ -92,6 +93,11 @@ impl ScalarUDFImpl for SparkLengthFunc {
     fn aliases(&self) -> &[String] {
         &self.aliases
     }
+
+    fn return_field_from_args(&self, args: ReturnFieldArgs) -> Result<FieldRef> {
+        let nullable = args.arg_fields.iter().any(|f| f.is_nullable());
+        Ok(Arc::new(Field::new(self.name(), Boolean, nullable)))
```

Comment:
this function does not return a boolean but Int32, see the `return_type` function above.

can you please also add a test that it is nullable when the child is nullable